src/riscv.rs

```diff
@@ -0,0 +1,16 @@
+// Implementation from gcc
+// https://raw.githubusercontent.com/gcc-mirror/gcc/master/libgcc/config/epiphany/mulsi3.c
+#[cfg_attr(not(feature = "mangled-names"), no_mangle)]
+pub unsafe extern "C" fn __mulsi3(mut a: usize, mut b: usize) -> usize {
```

Comment:
The signature should be `fn(u32, u32) -> u32` to match the linked source. (iirc, "si" is used for 32-bit ops; "di" is used for 64-bit ops; etc.).